File: src/main/java/org/jcodec/codecs/h264/io/model/PredictionWeightTable.java
package org.jcodec.codecs.h264.io.model;

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 * 
 * @author The JCodec project
 * 
 */
public class PredictionWeightTable {
    // luma_log2_weight_denom
    public int lumaLog2WeightDenom;
    // chroma_log2_weight_denom
    public int chromaLog2WeightDenom;

    // luma_weight
    public int[][] lumaWeight;
    // chroma_weight
    public int[][][] chromaWeight;

    // luma_offset
    public int[][] lumaOffset;
    // chroma_offset
    public int[][][] chromaOffset;

    public PredictionWeightTable() {
        this.lumaWeight = new int[2][];
        this.chromaWeight = new int[2][][];

        this.lumaOffset = new int[2][];
        this.chromaOffset = new int[2][][];
    }
}

File: src/main/java/org/jcodec/codecs/h264/io/model/PredictionWeightTable.kt
package org.jcodec.codecs.h264.io.model

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 *
 * @author The JCodec project
 */
class PredictionWeightTable {
    // luma_log2_weight_denom
    @JvmField
    var lumaLog2WeightDenom = 0

    // chroma_log2_weight_denom
    @JvmField
    var chromaLog2WeightDenom = 0

    // luma_weight
    @JvmField
    var lumaWeight: Array<IntArray?>

    // chroma_weight
    @JvmField
    var chromaWeight: Array<Array<IntArray>?>

    // luma_offset
    @JvmField
    var lumaOffset: Array<IntArray?>

    // chroma_offset
    @JvmField
    var chromaOffset: Array<Array<IntArray>?>

    init {
        lumaWeight = arrayOfNulls(2)
        chromaWeight = arrayOfNulls(2)
        lumaOffset = arrayOfNulls(2)
        chromaOffset = arrayOfNulls(2)
    }
}